extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/devmode/KeyPairOutOfTheBoxDevModeTest.java

```diff
@@ -0,0 +1,83 @@
+package io.quarkus.jwt.test.devmode;
+
+import io.quarkus.jwt.test.GreetingResource;
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+import io.restassured.http.Header;
+import io.smallrye.jwt.build.Jwt;
+import jakarta.annotation.security.PermitAll;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import org.eclipse.microprofile.jwt.Claims;
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+
+public class KeyPairOutOfTheBoxDevModeTest {
+
+    @RegisterExtension
+    static QuarkusDevModeTest devMode = new QuarkusDevModeTest()
+            .withApplicationRoot((jar) -> jar.addClasses(
+                    TokenResource.class,
+                    GreetingResource.class).addAsResource(
+                            new StringAsset(""), "application.properties"));
+
+    @Test
+    void shouldNotBeNecessaryToAddSignKeysOnApplicationProperties() {
+        String token = RestAssured.given()
+                .header(new Header("Accept", "text/plain"))
+                .get("/token")
+                .andReturn()
+                .body()
+                .asString();
+
+        RestAssured.given()
+                .header(new Header("Authorization", "Bearer " + token))
+                .get("/only-user")
+                .then().assertThat().statusCode(200);
+    }
+
+    @Test
+    void shouldUseTheSameKeyPairOnLiveReload() {
+        String token = RestAssured.given()
+                .header(new Header("Accept", "text/plain"))
+                .get("/token")
+                .andReturn()
+                .body()
+                .asString();
+
+        devMode.modifySourceFile("GreetingResource.java", s -> s.replace("Hello from Quarkus", "Hello from JWT"));
+
+        // there is no need to get another token
+        RestAssured.given()
+                .header(new Header("Authorization", "Bearer " + token))
+                .get("/only-user")
+                .then().assertThat().statusCode(200)
+                .body(Matchers.containsString("Hello from JWT"));
+    }
+
+    @Path("/token")
+    static class TokenResource {
+
+        @GET
+        @Produces(MediaType.TEXT_PLAIN)
+        @PermitAll
+        public String hello() {
+            return Jwt.issuer("https://example.com/issuer")
```

Comment:
Yeah, this is interesting... I think we should set the `issuer` properties (for `smallrye-jwt` and `smallrye-jwt-build`) as well if they are not set (if at least one of them is set, then we don't do it...), as the `issuer` is a technical claim...

Reply:
@mcruzdev I'm not 100% sure, if we have users who are only starting with smallrye-jwt and devmode, then they likely have no idea yet which provider will issue tokens, they just want to do a quick devmode try... In this case asking users to set `https://example.com/issuer` is no more dev friendly than just setting it out of the box

Reply:
@mcruzdev It is required for `smallrye-jwt`, so if we don't set it in devmode to something like `https://example.com/issuer` then users who are only starting will have to type some random string anyway - which won't work in prod. 
Those users which know the issuer values will type them straight away, we won't set it in devmode if this property is configured